pkg/controllers/argocd.go

```diff
@@ -0,0 +1,42 @@
+package controllers
+
+import (
+	"context"
+	"crypto/tls"
+	"net/http"
+
+	argocd "github.com/stefanmcshane/go-argocd/argocd"
+	"github.com/stefanmcshane/go-argocd/argocd/client/application_service"
+)
+
+type ArgoCD struct {
+	// Client argocdclient.
+}
+
+func NewArgoCDClient() (ArgoCD, error) {
```

Comment:
Catchall function for now, which would be broken up into:
- `NewArgoCDClient()` which takes the config required as a param and returns the client
- Convenience functions for reading whatever data we need from the ArgoCD server